src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -621,6 +621,11 @@ public void OnTransactionCommitted(LowLevelTransaction tx)
                 action?.Invoke(tx);
             }
 
+            public void OnTransactionCompleted()
+            {
+               _waitForJournalStateUpdateUnderTx.Set();
```

Comment:
Previously this was called under `PreventNewTransactions()` lock. Need to think about implications of calling it from here.

Reply:
This MRE was created and supposed to be called when we "update the journal state" - this is no longer the case so now it's misleading.

Also we are no longer under write tx lock when this is called. This might be important.

Reply:
If we are under the write tx lock, we can update the journals.
If we aren't, we need to let ourselves acquire it in order to update the journals.

Reply:
Ignore my above comments about `PreventNewTransactions()` and being called under write tx lock. I took a wrong look at the code.

But see my below suggestion about using another MRE to cover this scenario.